### PR TITLE
Generate rust documenation

### PR DESCRIPTION
**Pull request by boxdot**
For the detailed list of things that were changed, see the individual commits.

### CODE REVIEW

File: flatdata-generator/flatdata/generator/grammar.py [hunk 1]
```diff
@@ -185,6 +187,8 @@ def _combine_list(t):
     Optional(comment)
 )
 
-flatdata_grammar = Group(free_comments +
-                         OneOrMore(namespace)("namespace")
-                         )("flatdata")
+flatdata_grammar = Group(
+    Optional(comment)("doc") +
```
Comment (VeaaC) [inline flatdata-generator/flatdata/generator/grammar.py]: Not sure about this one: Most languages do not support this... and then were does it end up?

File: flatdata-generator/flatdata/generator/templates/rust/structure.jinja2 [hunk 2]
```diff
@@ -56,11 +83,6 @@ impl<'a> {{ struct.name }}Ref<'a> {
 
     {% endif %}
     {% endfor %}
-
```
Comment (VeaaC) [inline flatdata-generator/flatdata/generator/templates/rust/structure.jinja2]: Unrelated to change?

File: flatdata-generator/flatdata/generator/templates/rust/structure.jinja2 [hunk 3]
```diff
@@ -110,23 +143,14 @@ impl<'a> {{ struct.name }}Mut<'a> {
...
 
+    /// Copies the data from `other` into this struct.
     #[inline]
     pub fn fill_from(&mut self, other: &{{ struct.name }}Ref) {
         {% for field in struct.fields %}
         {% set name = field.name | escape_rust_keywords %}
         self.set_{{ name }}(other.{{ name }}());
         {% endfor %}
     }
-
```
Comment (VeaaC) [inline flatdata-generator/flatdata/generator/templates/rust/structure.jinja2]: Unrelated to change?

File: flatdata-generator/tests/generators/rust_expectations/archives/comments.rs [hunk 4]
```diff
@@ -0,0 +1,232 @@
+// Do not edit: This code was generated by flatdata's generator.
```
Comment (VeaaC) [inline flatdata-generator/tests/generators/rust_expectations/archives/comments.rs]: remove all non-relevant code (e.g. the schemas?)

File: flatdata-generator/flatdata/generator/templates/rust/rust.jinja2 [hunk 5]
```diff
@@ -8,8 +8,16 @@
 {%- for ns in tree.root.children recursive %}
 {%- if ns | is_namespace %}
 
+{% if ns | is_builtin_namespace %}
+#[doc(hidden)]
```
Comment (VeaaC) [inline flatdata-generator/flatdata/generator/templates/rust/rust.jinja2]: Nice...

I have also been thinking: Most likely we should move the schemas to the bottom of the file... they not the least useful thing to rread

File: flatdata-generator/flatdata/generator/grammar.py (hunk 1 above)
Comment (boxdot) [reply flatdata-generator/flatdata/generator/grammar.py]: Ok, removed.

File: flatdata-generator/flatdata/generator/templates/rust/structure.jinja2 (hunk 2 above)
Comment (boxdot) [reply flatdata-generator/flatdata/generator/templates/rust/structure.jinja2]: Actually it is. It removed dead code in Rust.

File: flatdata-generator/tests/generators/rust_expectations/archives/comments.rs (hunk 4 above)
Comment (boxdot) [reply flatdata-generator/tests/generators/rust_expectations/archives/comments.rs]: Done.

File: flatdata-generator/flatdata/generator/templates/rust/rust.jinja2 (hunk 5 above)
Comment (boxdot) [reply flatdata-generator/flatdata/generator/templates/rust/rust.jinja2]: Makes sense. I will do in the next PR, since otherwise the diff will become crazy. https://github.com/heremaps/flatdata/issues/168

**Comment by VeaaC** [comment]
Merge?
